src/store/notifications/saga.ts

```diff
@@ -21,13 +31,81 @@ export function* fetch(action) {
   yield put(setStatus(AsyncListStatus.Idle));
 }
 
+export function createEventChannel(userId, pusherClient = new PusherClient()) {
+  return eventChannel((emit) => {
+    const events = pusherClient.events.map((event) => {
+      return {
+        key: event,
+        callback: (notification) => {
+          // ToDo: return user data in notification
+          emit({ ...notification, isUnread: true });
+        },
+      };
+    });
+
+    pusherClient.init(userId, events);
+
+    const unsubscribe = () => {
+      pusherClient.disconnect();
+    };
+
+    return unsubscribe;
+  });
+}
+
+export function* watchForChannelEvent(userId) {
+  const notificationChannel = yield call(createEventChannel, userId);
+
+  while (true) {
+    const { abort, notification } = yield race({
+      abort: take(SagaActionTypes.CancelEventWatch),
+      notification: take(notificationChannel),
+    });
+
+    if (abort) {
+      notificationChannel.close();
+      return false;
+    }
+
+    if (relevantNotificationTypes.includes(notification.notificationType)) {
+      yield call(addNotification, notification);
+    }
```

Comment:
I think we just add every notification? The denormalize already filters all the event types.